### ISSUE TITLE
Bug: Text format lost after auto-correcting on iOS

### ISSUE THREAD

**Issue by samirant15** [facebook/lexical#4801]
On an IOS device, if the user typed a misspelled word with formats like Bold or Italic, after clicking the autocorrected work on the native keyboard, it replaces the word with the correct one but the formats are lost.

Example from lexical playground website and an IOS simulator (note: it also happens on physical devices)
![autocorrect-lexical](https://github.com/facebook/lexical/assets/18222884/81204027-f7b4-466a-99d9-7ca9f3cdd421)

Lexical version: 0.11.2

## Steps To Reproduce

1. On a lexical editor type a word with formatting applied
2. Make sure it is a misspelled word so the native keyboard prompts you to correct it
3. Tap on the native keyboard suggestion and the word will be corrected but the formats are lost

Link to code example: It happens in the playground https://github.com/facebook/lexical/tree/main/packages/lexical-playground (see above screen record)

## The current behavior
After autocorrecting a word on IOS the text formats are lost

## The expected behavior
After autocorrecting a word on IOS the text formats should be kept

**Comment by cakj** [comment]
Hi,
it's not only for iOS devices, you can reproduce this issue on a Mac with safari as well.

**Comment by laulautrup** [comment]
Yes we also experience this issue on iOS version 0.12.2. Did anyone find a fix yet?

**Comment by amanharwara** [comment]
@acywatson I don't think this is an issue with composition. From what I can tell, the trace looks something like this:
`onBeforeInput` -> `insertReplacementText` -> `CONTROLLED_TEXT_INSERTION_COMMAND` -> `$insertDataTransferForRichText` -> `$insertGeneratedNodes` -> `RangeSelection.insertNodes` -> `removeTextAndSplitBlock`

Maybe the inserted nodes should be given the format of the current selection? 🤔

**Comment by amanharwara** [comment]
This issue also doesn't happen if we ignore the `text/html` string in the `$insertDataTransferForRichText` and only use the `text/plain` content.

So one possible fix might be to have the `event` be one of the parameters for the `insertDataTransfer` function, and if the `event.input.type === "insertReplacementText"` then ignore the `text/html` data.
